src/turbolinksAdapter.ts

```diff
@@ -19,12 +21,33 @@ export class TurbolinksAdapter {
     if (response == null) return
     if (response.redirected != true) return
 
-    const action = 'advance'
+    // const action = 'advance'
 
     // TODO: When should we actually use replace other than when specified??
+    // leastbad says: this should be advance unless data-turbolinks-action="replace"
     //   action = 'replace'
 
-    window.Turbolinks.clearCache()
-    window.Turbolinks.visit(location, { action })
+    // i think that this should only happen if data-ujs-morph has been set, otherwise default to Turbolinks visit
+    this.body(event).then(html => {
+      const template = document.createElement('template')
+      template.innerHTML = String(html).trim()
+      morphdom(document.body, template.content, {childrenOnly: true})
+      
+      window.history.pushState({}, '', response.url)
```

Comment:
Im not even 100% we should pushState. Is it necessary? since the person never technically redirects and only makes a POST request? Idk, kinda thinking out loud here.

Reply:
It's necessary if there's a redirect on the table. eg when I login, we don't want them looking at the logged-in UI but seeing a /users/sign_in path.